src/StreamJsonRpc/JsonRpc.cs

```diff
@@ -1042,6 +1042,32 @@ public void Dispose()
             GC.SuppressFinalize(this);
         }
 
+        /// <summary>
+        /// Tries to obtain the <see cref="MethodSignatureAndTarget"/> info of a method.
+        /// </summary>
+        internal bool MethodSupportsSingleParameterObjectDeserialization(string methodName, Type parameterType)
```

Comment:
Being internal means this magic trick isn't reproducible by other (external) formatters. I like that it's your default, but I'd like to explore whether we can make the functionality available publicly. 
Your method as it stands is very focused on your feature. If however we change it to simply return the `JsonRpcMethodAttribute` for *any* RPC method, then it will likely be useful not just for this but other future features, which makes me feel better about making it public.
I'm going to play around with this idea and may push a commit to your PR to see what you think, if that's OK with you.